### ISSUE TITLE
Evil-goggles should not be on by default (or part of core spacemacs)

### ISSUE THREAD

**Issue by quicknir** [syl20bnr/spacemacs#10834]
This is a very bizarre thing to have on by default, and in a core layer. Packages that emulate much more popular vim packages, like evil-snipe, are bundled in their own separate layer. Spacemacs core for the most part sticks to evil packages that are either required to emulate things built in with vim (like evil numbers), or evil packages that are extremely popular with the vim community (like surround).

If you can argue in favor of evil-goggles, there are a dozen more evil packages that other people can argue in favor of, that will all end up in core spacemacs and on by default. And things inevitably end up being bloated. There's an extremely simple solution for this, which is just to put evil-goggles into a well written layer, like evil-snipe. I use evil-snipe and I would never argue for its inclusion into core. 

Evil-goggles is actually worse than many packages to have included and on by default, because unlike many other packages (e.g. evil-mc), it's not the kind of thing that you don't notice if you don't use. It's up and in your face from the first second you start using spacemacs. It's going to be a turn-off for as many people as enjoy it.

It's a nice little opinionated package, which is great, but such packages shouldn't be on by default nor even in the default distribution layers; it should be in a layer on its own.

Note that #9511 can be closed.

**Comment by syl20bnr** [comment]
I don't agree, I made it a core package on purpose because it adds
useful visual feedback about objects. This is in the same league as
quick highlight for evaluated code in Emacs Lisp or Clojure.

Now we can add a dotspacemacs variable to control it, maybe as a
opt-out variable of the editing style like hybrid already has.

**Comment by quicknir** [comment]
Useful for you. Many people (myself included) find it quite distracting. Again, I'm not sure why such an opinionated package is on by default? It would be like including evil-snipe with multiline `f`/`t` by default. Yes, I find it infinitely more useful than the default, it's hard for me to understand why anyone would prefer the default, and yet I'm willing to bet that many people would not like evil-snipe.

What makes it worse is that this is much more trivial for a beginner to disable, as it stands, then it would be for a beginner to enable it, if it were a separate layer. A dot-spacemacs variable addresses this, but again: dot-spacemacs already has many many variables that people rarely go through in full detail.

If you really want this on by default in spacemacs, having it in a separate layer is *still* a better solution. Just put the layer in the default .spacemacs file. Then it is easy to remove, or add, for beginners, without bloating the .spacemacs options. And it only takes up space if you are actually using it. As a bonus, it also keeps the codebase more modular.

**Comment by quicknir** [comment]
I'm not sure what else an apt comparison is, because plugins of this type are so rare. I could point out that even *line numbers*, used by infinitely more programmers than this, is off by default. But then you could say, but line numbers take up space. Etc. I could ask why we don't have indentation guides of any kind on by default, despite the fact that this is actually a common thing to do across many editors, instead of a rather odd one-of-a-kind feature, but I imagine you'll have a reason for that too. There will never be an exact 1-1 comparison. I compared to evil-snipe because they are both evil plugins that are not part of the vanilla vim experience, that appeal to a minority of people, and there are no complications involved in leaving them out of core.

I was actually going to leave this in my personal setup (even though I still don't think it should be in core), until I tested `cc`. Because with `c` commands you usually want to start typing in the exact space you just deleted, not only is it incredibly distracted having flashing text in the exact space you're trying to type, but it's noticeably slower as well.

Muscle memory or not it's very obvious there's going to be roughly as many people who severely dislike this, as like it. Most of the people I know who use vim, if I put them in front of spacemacs, would ask me how to disable this as among their first questions. Packages that irritate a fraction of users, even if it is a significant minority rather than a majority, shouldn't be in and on by default; it doesn't take many packages like that before the out of the box experience is annoying for everyone.

**Comment by ljupchokotev** [comment]
I have to agree with @quicknir on this. Putting it as a layer that is enabled by default in the .spacemacs template should be enough. It's easy to disable if you don't want it and it's still the default option.

I haven't seen anything like this in other editors and it also makes spacemacs appear slower than it actually is, at least for me.

**Comment by fiveNinePlusR** [comment]
that's a fair compromise even though i use evil-goggles and like that functionality.

**Comment by Miciah** [comment]
Can't evil-goggles be disabled by adding `evil-goggles` to `dotspacemacs-excluded-packages`?

I find evil-goggles useful for things like yanking where there is otherwise no visual feedback.  However, I disabled it a while back because it was sometimes so slow, I could get several commands ahead of Emacs's screen updates.  In such situations, the performance drag of evil-goggles is disorienting to such a degree that it *does* interfere with muscle memory.  I see that Spacemacs disables pulsing and lowers shortens evil-goggles's highlight duration, so I'll try it again and see how it feels. 

Would those opposed to evil-goggles be less opposed if it were enabled only for a smaller set of commands by default? For example, disable `evil-goggles-enable-change` and `evil-goggles-enable-delete` by default.

**Comment by syl20bnr** [comment]
Alright I disabled it by default.

I moved the `dotspacemacs-xxxx` variables related to Vim style to
`vim-style-xxxx` and created a new variable
`vim-style-enable-visual-feedback` which defaults to nil.

I also went more aggressive on the blink timings, the idea is to have a
quick feedback without slowing the editing experience.

Variables are documented here: https://github.com/syl20bnr/spacemacs/blob/develop/doc/DOCUMENTATION.org#vim
